rhine-terminal/TerminalSimple.hs

```diff
@@ -0,0 +1,113 @@
+{- | Example application for the @rhine-terminal@ library. -}
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+-- base
+import Prelude hiding (putChar)
+import System.Exit (exitSuccess)
+import System.IO hiding (putChar)
+import Control.Monad
+
+-- text
+import Data.Text (Text)
+import qualified Data.Text as T
+-- terminal
+import System.Terminal
+import System.Terminal.Internal
+-- rhine
+import FRP.Rhine
+-- rhine-terminal
+import FRP.Rhine.Terminal
+
+type App = TerminalT LocalTerminal IO
+
+-- Clocks
+
+data Input
+  = Char Char Modifiers
+  | Space
+  | Backspace
+  | Enter
+  | Exit
+
+type InputClock = SelectClock TerminalEventClock Input
+
+inputClock :: InputClock
+inputClock = SelectClock
+  { mainClock = TerminalEventClock
+  , select }
+    where
+      select :: Tag TerminalEventClock -> Maybe Input
+      select = \case
```

Comment:
In fact if you use `\case`, you can define it directly without a `where` clause! I like that a lot. I'll use `\case` more in my own code I think.